### PR TITLE
[MNT] Pandas 2.0.0 deprecation actions (part 2)

### PR DESCRIPTION
**Pull request by aiwalter**
succesor PR of #153 

issue: #150

### CODE REVIEW

**Comment by aiwalter** [comment]
@MatthewMiddlehurst could you pls have a look at this test? `aeon/transformations/panel/tests/test_interpolate.py::test_resizing`, its failing for `pandas==2.0.0rc0`.

**Comment by MatthewMiddlehurst** [comment]
I can probably have a look at it at some point if needed, but I have not ever touched those tests or the transformer.

File: aeon/forecasting/base/adapters/_statsforecast.py [hunk 1]
```diff
@@ -24,8 +24,8 @@ class _StatsForecastAdapter(BaseForecaster):
         "requires-fh-in-fit": False,  # is forecasting horizon already required in fit?
         "X-y-must-have-same-index": False,  # can estimator handle different X/y index?
         "enforce_index_type": None,  # index type that needs to be enforced in X/y
-        "capability:pred_int": True,  # does forecaster implement predict_quantiles?
-        "python_dependencies": "statsforecast",
+        # "capability:pred_int": True,  # does forecaster implement predict_quantiles?
```
Comment (aiwalter) [inline aeon/forecasting/base/adapters/_statsforecast.py:27]:
```suggestion
        "capability:pred_int": True,  # does forecaster implement predict_quantiles?
```

**Comment by aiwalter** [comment]
@MatthewMiddlehurst  ok the test `aeon/transformations/panel/tests/test_interpolate.py::test_resizing` seems actually to fail with `pandas==2.0.0rc0`

**Review by GuzalBulatova — APPROVED** [review]
LGTM, great job @aiwalter !

**Pull request merged**